### DISCUSSION TITLE
Using next/font in a TypeScript library used by a Next.js application

### DISCUSSION THREAD

**Discussion by callajd** [vercel/next.js#66053] [Help]
### Summary

Hi!

I have two projects:
1. TypeScript library that implements a few UI components, custom react hooks, etc.  
2. Next.js web application (Next.js 14.2.3 / AppDir) that depends on library above

One of the UI components in this library uses a font from `next/font/google`:

```
import { Barlow_Semi_Condensed } from 'next/font/google';

const LogoFont = Barlow_Semi_Condensed({
  display: 'swap',
  subsets: ['latin'],
  weight: ['400', '500', '600', '700', '800', '900'],
  style: ['normal', 'italic']
})
```

When I build the web application, I get the following error:

```
Failed to compile.

./node_modules/@actuate-ag/auth/dist/client/index.js
Error: 
  x Font loader calls must be assigned to a const
    ,-[/Users/jefferycallahan/code/actuate-web-server/node_modules/@actuate-ag/auth/dist/client/index.js:72:1]
 72 |     // src/client/components/authCard.tsx
 73 |     import { Barlow_Semi_Condensed } from "next/font/google";
 74 |     import { Fragment, jsx as jsx2, jsxs } from "react/jsx-runtime";
 75 | ,-> var LogoFont = Barlow_Semi_Condensed({
 76 | |     display: "swap",
 77 | |     subsets: ["latin"],
 78 | |     weight: ["400", "500", "600", "700", "800", "900"],
 79 | |     style: ["normal", "italic"]
 80 | `-> });
 81 |     function AuthCard({ loading, footer: footer3 = () => /* @__PURE__ */ jsx2(Fragment, {}), children }) {
 82 |       return /* @__PURE__ */ jsxs("div", { className: styles_default.formOutline, children: [
 83 |         /* @__PURE__ */ jsx2(LoadingOverlay, { visible: loading, overlayProps: { blur: 2 } }),
    `----
```

I should note that I'm a backend engineer by trade and not intimately familiar with all of the nuances of JS/TS bundling formats.  I can't figure out how to fix this error.  I assume it's related to how I'm transpiling and packaging the TypeScript library, but after trying many different changes, I'm stuck.

Any advice on how to configure the library so I can successfully use it in my Next.js application?  If it's helpful, I can share my library package.json / tsconfig.json, let me know.

Thanks and I appreciate it!

**Comment by tklives** [comment]
I am having the same issue - I am defining this as a const in my ui package, but when I attempt to use the fonts in the consuming application (storybook in my case), I am getting the same error about needing to be a const.

What could be causing this to get switched to a var somewhere along the way?

**Reply by tklives** [reply]
I have an update on the issue I was having.  (@callajd )

In the `main.ts` file, if I remove `webpackFinal:` from the StorybookConfig, everything functions as expected.  

I'm not sure why storybook for next ships with the webpack configuration, but removing it was my ticket.  Hope this helps someone :)

**Comment by Ravi-TSI** [comment]
It is same issue for me too, anyone fixed
 `Failed to compile.

./node_modules/@actuate-ag/auth/dist/client/index.js
Error: 
  x Font loader calls must be assigned to a const
 73 |     import { Barlow_Semi_Condensed } from "next/font/google";
 74 |     import { Fragment, jsx as jsx2, jsxs } from "react/jsx-runtime";
 75 | ,-> var LogoFont = Barlow_Semi_Condensed({
 76 | |     display: "swap",
 77 | |     subsets: ["latin"],
 78 | |     weight: ["400", "500", "600", "700", "800", "900"],
 79 | |     style: ["normal", "italic"]
 80 | `-> });`

**Reply by HasanMothaffar-OSG** [reply]
For those coming in late, the issue is related to `browserslist` config. I had this config in my package.json (I don't know why, maybe it was there when I created the app):

```json
 "browserslist": {
        "production": [
            ">0.2%",
            "not dead",
            "not op_mini all"
        ],
        "development": [
            "last 1 chrome version",
            "last 1 firefox version",
            "last 1 safari version"
        ]
    },
```

If you modify this field, it works. There might be something wrong with the config, I didn't try to modify values in it.
